test/migrator_utils.dart

```diff
@@ -37,24 +29,33 @@ void testMigrator(String migrator) {
 /// Run the migration test in [hrxFile].
 ///
 /// See migrations/README.md for details.
-_testHrx(File hrxFile, String migrator) async {
+Future<void> _testHrx(File hrxFile, String migrator) async {
   var files = _HrxTestFiles(hrxFile.readAsStringSync());
   await files.unpack();
-  Map<Uri, String> migrated;
-  var entrypoints =
-      files.input.keys.where((path) => path.startsWith("entrypoint"));
-  var arguments = [migrator]..addAll(files.arguments)..addAll(entrypoints);
-  await expect(
-      () => IOOverrides.runZoned(() async {
-            migrated = await MigratorRunner().run(arguments);
-          }, getCurrentDirectory: () => Directory(d.sandbox)),
-      prints(files.expectedLog?.replaceAll("\$TEST_DIR", d.sandbox) ?? ""));
-  var importer = FilesystemImporter(d.sandbox);
-  for (var file in files.input.keys) {
-    expect(migrated[importer.canonicalize(Uri.parse(file))],
-        equals(files.output[file]),
-        reason: 'Incorrect migration of $file.');
+
+  var process = await TestProcess.start(
+      Platform.executable,
+      [
+        "--enable-asserts",
+        p.absolute("bin/sass_migrator.dart"),
+        migrator,
+        '--no-unicode'
+      ]
+        ..addAll(files.arguments)
+        ..addAll(
+            files.input.keys.where((path) => path.startsWith("entrypoint"))),
+      workingDirectory: d.sandbox,
+      description: "migrator");
+
+  if (files.expectedLog != null) {
+    expect(process.stdout,
+        emitsInOrder(files.expectedLog.trimRight().split("\n")));
   }
+  expect(process.stdout, emitsDone);
+  await process.shouldExit(0);
+
+  await Future.wait(files.output.keys
+      .map((path) => d.file(path, files.output[path]).validate()));
```

Comment:
We should also test any input files that aren't in the output to confirm that they are unchanged.

Reply:
Good call. Done.